src/onnx/parse_slice.cpp

```diff
@@ -77,7 +78,37 @@ struct parse_slice : op_parser<parse_slice>
             op.axes = axes;
         }
 
-        return info.add_instruction(op, args[0]);
+        std::vector<int64_t> raxes;
+        if(std::any_of(steps.begin(), steps.end(), [](auto s) { return s < 0; }))
+        {
+            for(auto i : op.axes)
```

Comment:
This should be `for(auto i : range(op.axes.size()))`. This should fix the UBSAN error.

Reply:
Undefined behavior is gone but now I am seeing ASAN problem:

```
SUMMARY: AddressSanitizer: heap-buffer-overflow (/opt/rocm/lib/libmigraphx_onnx.so.1+0x8a7c0a) 
Shadow bytes around the buggy address:
[...]
```

gdb:
```
addition of unsigned offset to 0x602000000dd0 overflowed to 0x602000000dc8
```

backtrace:
```
#6  0x00007ffff71b5c24 in migraphx::version_1::onnx::onnx_parser::parse_graph (this=<optimized out>, mod=<optimized out>, graph=...)
    at /code/AMDMIGraphX/src/onnx/onnx_parser.cpp:332
#7  0x00007ffff71b3c53 in migraphx::version_1::onnx::onnx_parser::parse_from (this=<optimized out>, is=..., Python Exception <class 'OverflowError'> int too big to convert: 
name=)
    at /code/AMDMIGraphX/src/onnx/onnx_parser.cpp:235
```

Reply:
Can you push your changes? I'll take a look tomorrow to see what the issue is.

Reply:
Done, thanks!